packages/core/src/controllers/ChainController.ts

```diff
@@ -0,0 +1,292 @@
+    key: K,
+    callback: (value: ChainControllerState[K]) => void
+  ) {
+    return subKey(state, key, callback)
+  },
+
+  subscribeChain(callback: (value: ChainAdapter | undefined) => void) {
+    let prev: ChainAdapter | undefined = undefined
+    const activeChain = state.activeChain || ConstantsUtil.CHAIN.EVM
+
+    return sub(state.chains, () => {
+      const nextValue = state.chains.get(activeChain)
+      if (!prev || prev !== nextValue) {
+        prev = nextValue
+        callback(nextValue)
+      }
+    })
+  },
+
+  subscribeChainProp<K extends keyof ChainAdapter>(
+    property: K,
+    callback: (value: ChainAdapter[K] | undefined) => void
+  ) {
+    let prev: ChainAdapter[K] | undefined = undefined
+    const activeChain = state.activeChain || ConstantsUtil.CHAIN.EVM
+
+    return sub(state.chains, () => {
+      const nextValue = state.chains.get(activeChain)?.[property]
+      if (prev !== nextValue) {
+        prev = nextValue
+        callback(nextValue)
+      }
+    })
+  },
+
+  initialize(adapters: ChainsInitializerAdapter[]) {
+    const firstChainToActivate = adapters?.[0]?.chain || ConstantsUtil.CHAIN.EVM
+
+    state.activeChain = firstChainToActivate
+
+    adapters.forEach((adapter: ChainsInitializerAdapter) => {
+      state.chains.set(adapter.chain, {
+        chain: adapter.chain,
+        connectionControllerClient: adapter.connectionControllerClient,
+        networkControllerClient: adapter.networkControllerClient,
+        accountState,
+        networkState
+      })
+    })
+  },
+
+  setMultiChainEnabled(multiChain: boolean) {
+    state.multiChainEnabled = multiChain
+  },
+
+  setChainNetworkData(chain: Chain | undefined, props: Partial<NetworkControllerState>) {
+    if (!chain) {
+      throw new Error('Chain is required to update chain network data')
+    }
+
+    const chainAdapter = state.chains.get(chain)
+
+    if (chainAdapter) {
+      chainAdapter.networkState = {
+        ...chainAdapter.networkState,
+        ...props
+      } as NetworkControllerState
+      state.chains.set(chain, chainAdapter)
+      NetworkController.replaceState(chainAdapter.networkState)
+    }
+  },
+
+  setChainAccountData(chain: Chain | undefined, accountProps: Partial<AccountControllerState>) {
+    if (!chain) {
+      throw new Error('Chain is required to update chain account data')
+    }
+
+    const chainAdapter = state.chains.get(chain)
+
+    if (chainAdapter) {
+      chainAdapter.accountState = {
+        ...chainAdapter.accountState,
+        ...accountProps
+      } as AccountControllerState
+      state.chains.set(chain, chainAdapter)
+      AccountController.replaceState(chainAdapter.accountState)
+    }
+  },
+
+  setAccountProp(
+    prop: keyof AccountControllerState,
+    value: AccountControllerState[keyof AccountControllerState],
+    chain?: Chain
+  ) {
+    this.setChainAccountData(state.multiChainEnabled ? chain : ConstantsUtil.CHAIN.EVM, {
```

Comment:
What about the Solana client? For user who implement Solana SDK, it should set the value to `solana` key by default, not only `ConstantsUtil.CHAIN.EVM`.